wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/MutableProjectConfig.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.project.server;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static com.google.common.collect.Maps.newHashMap;
+
+import com.google.common.annotations.Beta;
+import java.util.List;
+import java.util.Map;
+import org.eclipse.che.api.core.model.project.ProjectProblem;
+import org.eclipse.che.api.core.model.workspace.config.ProjectConfig;
+import org.eclipse.che.api.core.model.workspace.config.SourceStorage;
+import org.eclipse.che.api.project.shared.NewProjectConfig;
+import org.eclipse.che.api.workspace.shared.dto.CommandDto;
+
+/** @author Vlad Zhukovskiy */
+@Beta
+public class MutableProjectConfig implements ProjectConfig {
```

Comment:
I'm not an author of this class in your PR. Besides this class, you leave [this one](https://github.com/eclipse/che/blob/cb9a1112dd5d7e6564ba9991a98f621faa35def3/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/project/MutableProjectConfig.java) without changes. I'd propose to move `MutableProjectConfig` near `ProjectConfig` and not having two different implementations.

Reply:
I did not want to move the class to "shared" because it's not part of the protocolo between wsmaster and the IDE. Removed you as an author.

Reply:
As I see, file located is in `wsagent` module, so whats the problem to move implementation to the `che-core-api-project-shared` (which also is used by the client)? This will allow us not having two duplicated implementation where one of it is odd. You'll be able to use this class from your code and the client will be able to use it in own purposes.

Reply:
@tsmaeder What the problem move it to the `shared`? Project API is part of `wsagent` and not related to the `wsmaster` communications protocol